### ISSUE TITLE
 kubectl-gardenlogin process leakage

### ISSUE THREAD

**Issue by tedteng** [gardener/gardenctl-v2#275]
**What happened**:
kubelet command hung, and kubectl-gardenlogin process repeatedly generated
```
 501 85105 85104  0 10:06AM ttys000  0:00.01 kubectl-gardenlogin get-client-certificate
 501 85106 85105  0 10:06AM ttys000  0:00.01 kubectl-gardenlogin get-client-certificate
 501 85107 85106  0 10:06AM ttys000  0:00.01 kubectl-gardenlogin get-client-certificate
 501 85108 85107  0 10:06AM ttys000  0:00.01 kubectl-gardenlogin get-client-certificate
 501 85109 85108  0 10:06AM ttys000  0:00.01 kubectl-gardenlogin get-client-certificate
 501 85110 85109  0 10:06AM ttys000  0:00.01 kubectl-gardenlogin get-client-certificate
 501 85111 85110  0 10:06AM ttys000  0:00.01 kubectl-gardenlogin get-client-certificate
 501 85112 85111  0 10:06AM ttys000  0:00.01 kubectl-gardenlogin get-client-certificate
 501 85113 85112  0 10:06AM ttys000  0:00.01 kubectl-gardenlogin get-client-certificate
....

```
**What you expected to happen**:
```
`kubelet get xx` without issue
```
**How to reproduce it (as minimally and precisely as possible)**:
```
download `kubeconfig-gardenlogin`

gardenctl config set-garden abc --kubeconfig “<path-to-garden-cluster-kubeconfig>”

gardenctl target --garden abc

eval $(gardenctl kubectl-env zsh)

kubectl get xxx
```
**Anything else we need to know**:
observed non-admin user at the moment 

**Environment**:
```
gardenctl version
Version: [version.Info](http://version.info/){Major:“2”, Minor:“2", GitVersion:“v2.2.1”, GitCommit:“e235d0017d721bcf824266f7a4e2aaad78068282", GitTreeState:“clean”, BuildDate:“2023-02-17T12:43:08+00:00", GoVersion:“go1.19.6”, Compiler:“gc”, Platform:“darwin/arm64”}
```

**Comment by petersutter** [comment]
Hi @tedteng, you say `kubelet` but you probably mean `kubectl`, right?

In your steps on how to reproduce the issue, you download `kubeconfig-gardenlogin`. What do you mean by `kubeconfig-gardenlogin`. Where do you download it? Where is this file used later in your step-by-step description? 

> ```bash
> gardenctl config set-garden abc --kubeconfig “<path-to-garden-cluster-kubeconfig>”
> gardenctl target --garden abc
> eval $(gardenctl kubectl-env zsh)
> ```

here you target the garden `abc` with kubeconfig `<path-to-garden-cluster-kubeconfig>`. So the `kubectl get xxx` command that you run uses the kubeconfig that you set for your garden cluster, which is typically not a gardenlogin kubeconfig.

To wrap it up: With the given description I cannot see how a gardenlogin kubeconfig for a shoot cluster is used so that the `kubectl-gardenlogin get-client-certificate` command is executed.

**Comment by tedteng** [comment]
my bad yes `Kubectl`,  the steps I observed from my colleagues' laptop she downloads Kubeconfig from the dashboard, then put on somewhere as 
`<path-to-garden-cluster-kubeconfig>` and another case from an internal user ticket but from Linux with the same symptom


<img width="514" alt="image" src="https://user-images.githubusercontent.com/42234376/232943794-6e2030f6-32d2-453d-ae2d-1e6e6d0b13e6.png">

**Comment by petersutter** [comment]
Hi @tedteng, I think there are some confusions around the different kubeconfigs:
- Garden cluster kubeconfig: You can find this kubeconfig on the `MY ACCOUNT` page.
- Gardenlogin kubeconfig: This is the kubeconfig for the respective shoot cluster. Usually you would not download this kubeconfig manually from the dashboard if you are using gardenctl, as gardenctl generates this transparently for you.

You can also click on the (i) button to see the installation instructions where all steps are mentions (especially note the last sentence)

<img width="898" alt="03-gardenlogin-info" src="https://user-images.githubusercontent.com/5526658/233007608-89edbe0c-6911-4302-aae9-9f087cb63a82.png">

TL;DR: Do NOT configure the gardenlogin kubeconfig as garden cluster kubeconfig for gardenctl. The garden cluster kubeconfig can be downloaded from the `MY ACCOUNT` page on the dashboard
